packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -1955,6 +1964,15 @@ class _TimePickerDialogState extends State<TimePickerDialog> with RestorationMix
     _announceModeOnce();
   }
 
+  @override
+  void initState() {
+    super.initState();
+    if (widget.onEntryModeChanged != null) {
+      _entryModeListener = () => widget.onEntryModeChanged!(_entryMode.value);
+      _entryMode.addListener(_entryModeListener!);
+    }
```

Comment:
If we are going to check if `onEntryModeChanged` is not null here, we would need to do this again in the `didChangeDependencies` as the app could change this value on the next build to something different. For example, if they changed it from null to non-null, there would be no listener setup for it because it as null during `initState`.

However, rather than adding this same code block to `didChangeDependencies`, I think this could be simplified to just:

```dart
      _entryModeListener = () => widget.onEntryModeChanged?.call(_entryMode.value);
      _entryMode.addListener(_entryModeListener!);
```

That way we don't need to track changes as it will always just try to notify whatever listener is there if the mode changes.

In this case we just always create and add the listener, which means the listener could just be:

```dart
    late final VoidCallback _entryModeListener;
```

Reply:
Makes sense!